realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -287,6 +289,40 @@ public boolean isManaged() {
     }
 
     /**
+     * Gets {@link RealmConfiguration} from this object.
+     *
+     * @return {@link RealmConfiguration} of the Realm where this object belongs to or {@code null} if unmanaged.
+     * @throws IllegalStateException if this object was already deleted from Realm or the corresponding Realm of this object was already closed.
+     */
+    public RealmConfiguration getConfiguration() {
+        return RealmObject.getConfiguration(this);
+    }
+
+    /**
+     * Gets {@link RealmConfiguration} from managed {@link RealmModel} instance such as {@link RealmObject} or {@link DynamicRealmObject}.
+     *
+     * @param model managed Realm object.
+     * @return {@link RealmConfiguration} of the Realm where the {@code model} belongs to or {@code null} if the {@code model} is unmanaged.
+     * @throws IllegalArgumentException if the {@code model} is {@code null}.
+     * @throws IllegalStateException if the model was deleted or the corresponding Realm of the model was already closed.
+     */
+    public static RealmConfiguration getConfiguration(RealmModel model) {
+        if (model == null) {
+            throw new IllegalArgumentException(MSG_NULL_OBJECT);
+        }
+        if (!(model instanceof RealmObjectProxy)) {
```

Comment:
We need to return `null` when the object is managed even if the object is valid.